core/domain/classifier_services.py

```diff
@@ -83,8 +83,13 @@ def classify_string_classifier_rule(state, normalized_answer):
     best_matched_answer_group_index = len(state.interaction.answer_groups)
     best_matched_rule_spec_index = None
 
-    sc = classifier_registry.Registry.get_classifier_by_id(
-        feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput'])
+    if state.classifier_model_id == None:
+        sc = classifier_registry.Registry.get_classifier_by_id(
```

Comment:
Yes it does. The classifier_registry.Registry.get_classifier_by_id() refers to retrieving the LDAStringClassifier Class object.
The classifier_services.get_classifier_by_id() retrieves the Classifier Domain class.